lib/rest/Request.js

```diff
@@ -0,0 +1,207 @@
+"use strict";
+
+const Base = require("../structures/Base");
+const {fetch} = require("undici");
+
+const USER_AGENT = `DiscordBot (https://github.com/projectdysnomia/dysnomia, ${require("../../package.json").version})`;
+
+/**
+ * Represents the request.
+ */
+class Request {
+    /**
+     * The data to be set for the request body.
+     * @type {FormData | String}
+     */
+    data;
+
+    /**
+     * The RequestHandler.
+     * @type {RequestHandler}
+     */
+    handler;
+
+    /**
+     * The headers to attach to the request.
+     * @type {Object}
+     */
+    headers = {
+        "Accept-Encoding": "gzip,deflate",
+        "User-Agent": USER_AGENT
+    };
+
+    /**
+     * The major parameter of the request.
+     * @type {String}
+     */
+    majorParameter;
+
+    /**
+     * An uppercase HTTP method.
+     * @type {String}
+     */
+    method;
+
+    /**
+     * Data regarding the request.
+     * @type {Object}
+     */
+    options;
+
+    /**
+     * The endpoint to make the request to.
+     * @type {String}
+     */
+    path;
+
+    /**
+     * The route to make the request to.
+     * @type {String}
+     */
+    route;
+
+    /**
+     * The URL to make the request to.
+     * @type {URL}
+     */
+    url;
+
+    /**
+     * Represents the request.
+     * @arg {RequestHandler} handler Represents the RequestHandler.
+     * @arg {String} method An uppercase HTTP method.
+     * @arg {String} path The endpoint to make the request to.
+     * @arg {Object} options Data regarding the request.
+     * @arg {Boolean} [options.auth] Whether to add the "Authorization" header.
+     * @arg {Object} [options.body] The data to be set for the request body.
+     * @arg {Object} [options.headers] The headers to attach to the request.
+     * @arg {Object[]} [options.files] The files to attach to the request body.
+     * @arg {Boolean} [options.formdata] Whether to attach the body directly to form data instead of "payload_json" when sending files.
```

Comment:
nit: "form data" are two words
```suggestion
     * @arg {Boolean} [options.formData] Whether to attach the body directly to form data instead of "payload_json" when sending files.
```